src/main/java/edu/hm/hafner/analysis/parser/SimulinkCheckParser.java

```diff
@@ -0,0 +1,131 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.io.File;
+import java.io.IOException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * A parser for Simulink Check tool by Mathworks results.
+ * Used for .html report files.
+ *
+ * @author Eva Habeeb
+ */
+public class SimulinkCheckParser extends IssuePropertiesParser {
+
+    private static final long serialVersionUID = -8099258658775128275L;
+
+    @Override
+    @SuppressWarnings({"PMD.NcssCount", "PMD.CyclomaticComplexity", "PMD.CognitiveComplexity", "PMD.NPathComplexity" })
+    public Report parse(final ReaderFactory readerFactory) throws ParsingException {
+        String name = readerFactory.getFileName();
+        File file = new File(name);
+
+        try (IssueBuilder issueBuilder = new IssueBuilder()) {
+            //Parse html document using Jsoup
+            Document document = Jsoup.parse(file, "utf-8");
+            Elements systemElement = document.select("div.ReportContent");
+            String system = systemElement.first().id();
+            Report report = new Report();
+
+            //Select Warning items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.WarningCheck");
+            //Select Failed items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.FailedCheck");
+            //Select Not Run items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.NotRunCheck");
+            //Select Incomplete items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.IncompleteCheck");
+
+            return report;
+        }
+        catch (IOException e) {
+            throw new ParsingException(e);
+        }
+    }
+
+    /**
+     * Handles parsing.
+     * @param report the report object
+     * @param document parsed html document
+     * @param issueBuilder issueBuilder object
+     * @param system name of the system
+     * @param check category of the check
+     *
+     */
+    public void parseIssue(final Report report, final Document document, final IssueBuilder issueBuilder, final String system, final String check) {
+        Elements heading;
+        String issueTxt;
+        String headingElement;
+        String[] headingSplit;
+        int strLen;
+        String module;
+
+        Elements issueElements = document.select(check);
+        for (Element w : issueElements) {
```

Comment:
Are all other those HTML methods `null` safe? So what happens if not? IF the HTML document will be changed slightly then everything breaks with NPE.

Reply:
Yes the html methods used here are null safe, they return empty 'Element' objects if any of the check categories are not found in the report. All the HTML reports for simulink check have the same structure, the difference is that some reports might not have all the check categories, which is handled by the 'select' function.